core/lib/dal/src/transactions_dal.rs

```diff
@@ -571,7 +613,211 @@ impl TransactionsDal<'_, '_> {
         }
     }
 
-    async fn handle_executed_l2_transactions(
+    async fn insert_executed_l2_transactions(
```

Comment:
Dumb question: Would it be inefficient / impossible to rework queries updating executed transactions used by `mark_txs_as_executed_in_l2_block()` into insert-or-update queries? If this could be possible, it would lead to better code reuse and less queries to maintain.

Reply:
It's not possible with our schema. We have two unique constraints on `transactions`: hash and (initiator, nonce). Postgres allows upsert only on one unique constraint